pkg/controller/metric.go

```diff
@@ -10,13 +12,14 @@ const (
 )
 
 var (
-	errorGauge = prometheus.NewGauge(
-		prometheus.GaugeOpts{
+	errorCounterVec = prometheus.NewCounterVec(
```

Comment:
Counter makes more sense for error counting and we already have alerts using `rate()`.